src/mds/Server.cc

```diff
@@ -2584,6 +2609,12 @@ void Server::handle_client_request(const cref_t<MClientRequest> &req)
   if (session) {
     mdr->session = session;
     session->requests.push_back(&mdr->item_session_request);
+
+    if (has_completed && req->get_op() == CEPH_MDS_OP_CREATE) {
+      ceph_assert(created != inodeno_t());
+
+      set_reply_extra_bl(req, created, mdr->reply_extra_bl);
```

Comment:
I don't think this is the right place for this. It either needs to go in `handle_client_open` or in the switch statement in `Server::dispatch_client_request`.

Reply:
Okay, I will move it to the switch statement in `Server::dispatch_client_request()`. Thanks!